src/shogun2-core/src/test/java/de/terrestris/shogun2/rest/AbstractRestControllerTest.java

```diff
@@ -221,13 +224,19 @@ public void findById_shouldReturn_NotFound() throws Exception {
 	 */
 	@Test
 	public void save_shouldReturn_EntityAndCreated() throws Exception {
-		int id = 42;
+		final int id = 42;
 		String value = "save value";
 
 		TestModel withoutId = buildTestInstanceWithValue(value);
-		TestModel withId = buildTestInstanceWithIdAndValue(id, value);
 
-		when(serviceMock.saveOrUpdate(any(TestModel.class))).thenReturn(withId);
+		doAnswer(new Answer<Void>() {
+			@Override
+			public Void answer(InvocationOnMock invocation) throws Throwable {
+				TestModel entity = (TestModel) invocation.getArguments()[0];
+				IdHelper.setIdOnPersistentObject(entity, id);
+				return null;
```

Comment:
Is that the way how you fulfill the expected return type `void`? Can you change this to

```java
return;
```

or remove the `return` line entirely?

This would make this easier to read IMO.